server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -564,11 +568,15 @@ public void invariant() {
         }
     }
 
+    public boolean isInitialConfigurationSet() {
+        return getStateForMasterService().getLastAcceptedConfiguration().isEmpty() == false;
+    }
+
     public void setInitialConfiguration(final VotingConfiguration votingConfiguration) {
         synchronized (mutex) {
             final ClusterState currentState = getStateForMasterService();
 
-            if (currentState.getLastAcceptedConfiguration().isEmpty() == false) {
+            if (isInitialConfigurationSet()) {
```

Comment:
This is safe to extract because `getStateForMasterService` depends only on things protected by `mutex`.